### PR TITLE
Add `use_sts_region_from_client` to AWS Auth Config

### PR DESCRIPTION
**Pull request by raymonstah**
Adds a new field `use_sts_region_from_client` to the AWS auth config.

This feature will be available when https://github.com/hashicorp/vault/pull/21960 is released. Set for Vault `v1.15`.

### CODE REVIEW

File: vault/resource_aws_auth_backend_client.go [hunk 1]
```diff
@@ -69,6 +73,11 @@ func awsAuthBackendClientResource() *schema.Resource {
 				Optional:    true,
 				Description: "Region to override the default region for making AWS STS API calls.",
 			},
+			useSTSRegionFromClient: {
```
Comment (fairclothjm) [inline vault/resource_aws_auth_backend_client.go:76]: Should this be marked as Computed?

Comment (vinay-gopalan) [reply vault/resource_aws_auth_backend_client.go:76]: +1 on adding Computed to support the default :)

File: vault/resource_aws_auth_backend_client_test.go [hunk 2]
```diff
@@ -117,6 +117,27 @@ func TestAccAWSAuthBackendClientStsRegionNoEndpoint(t *testing.T) {
...
+	backend := acctest.RandomWithPrefix("aws")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			testutil.TestAccPreCheck(t)
+			SkipIfAPIVersionLT(t, testProvider.Meta(), provider.VaultVersion115)
+		},
+		Providers:    testProviders,
+		CheckDestroy: testAccCheckAWSAuthBackendClientDestroy,
+		Steps: []resource.TestStep{
+			{
```
Comment (vinay-gopalan) [inline vault/resource_aws_auth_backend_client_test.go:130]: Would be nice to add an initial step to test the default behavior and also ensuring that importing the field works 😄

**Review by vinay-gopalan — COMMENTED** [review]
Looking great! Could we also add docs for this field, thanks!

**Review by fairclothjm — APPROVED** [review]
LGTM!

File: website/docs/r/aws_auth_backend_client.html.md [hunk 3]
```diff
@@ -68,6 +68,11 @@ The following arguments are supported:
 * `sts_region` - (Optional) Override the default region when making STS API 
     calls. The `sts_endpoint` argument must be set when using `sts_region`.
 
+* `use_sts_region_from_client` - (Optional) If set, overrides both `sts_endpoint`
```
Comment (vinay-gopalan) [inline website/docs/r/aws_auth_backend_client.html.md:71]: Could we also document here that this field will only work with versions >= 1.15?

**Review by vinay-gopalan — APPROVED** [review]
Looks great! Had one small comment, but good to go otherwise 🎉

**Pull request merged**